mantidproject/mantid
# New stitch algorithm

**Description of work.**
Adds a new algorithm `Stitch` that meets the reactor-source LSS requirements, see the issue for more details why it's needed.

**To test:**

0. Build and read the documentation.

1. Create some workspaces with the script below as a template, and use the new algorithm to stitch. Try different options. Try to confuse it.
```
np.random.seed(179864)
x1 = np.linspace(0,100,11)
y1 = 5 + np.sin(x1) + np.random.normal(scale=0.001)
ws1 = CreateWorkspace(DataX=x1,DataY=y1,NSpec=1)
x2 = np.linspace(50,150,11)
y2 = 7 + np.cos(x2) + np.random.normal(scale=0.001)
ws2 = CreateWorkspace(DataX=x2,DataY=y2,NSpec=1)
st = Stitch([ws1,ws2])
```

2. Load the attached nexus file, use the first 3 workspaces in it to stitch. Try different options: reference name, manual and tied scale factors. Try the same input in different orders: increasing or decreasing distance, or completely random. Make sure that the output is what is expected and that the stitching is performed spectrum-wise. 

(remove the .zip from the name, it's a nexus file!)
[to_stitch_multi.nxs.zip](https://github.com/mantidproject/mantid/files/6883755/to_stitch_multi.nxs.zip)



Fixes #31055. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
